components/dashboard/src/teams/TeamUsageBasedBilling.tsx

```diff
@@ -0,0 +1,240 @@
+            clearTimeout(pollStripeCustomerTimeout!);
+        };
+    }, [pendingStripeCustomer, pollStripeCustomerTimeout, stripeCustomerId, team]);
+
+    if (!showUsageBasedUI) {
+        return <></>;
+    }
+
+    return (
+        <div className="mb-16">
+            <h3>Usage-Based Billing</h3>
+            <h2 className="text-gray-500">Manage usage-based billing, spending limit, and payment method.</h2>
+            <div className="max-w-xl">
+                <div className="mt-4 h-32 p-4 pt-6 pl-6 flex flex-col rounded-xl bg-gray-100 dark:bg-gray-800">
+                    <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Billing</div>
+                    {(isLoading || pendingStripeCustomer) && (
+                        <>
+                            <Spinner className="m-2 h-5 w-5 animate-spin" />
+                        </>
+                    )}
+                    {!isLoading && !pendingStripeCustomer && !stripeCustomerId && (
+                        <>
+                            <div className="text-xl font-semibold flex-grow text-gray-600 dark:text-gray-400">
+                                Inactive
+                            </div>
+                            <button className="self-end" onClick={() => setShowBillingSetupModal(true)}>
+                                Upgrade
+                            </button>
+                        </>
+                    )}
+                    {!isLoading && !pendingStripeCustomer && !!stripeCustomerId && (
+                        <>
+                            <div className="text-xl font-semibold flex-grow text-gray-600 dark:text-gray-400">
+                                Active
+                            </div>
+                            {/* <button className="self-end secondary" disabled={true}>
+                                Manage →
+                            </button> */}
+                        </>
+                    )}
+                </div>
+            </div>
+            {showBillingSetupModal && <BillingSetupModal onClose={() => setShowBillingSetupModal(false)} />}
+        </div>
+    );
+}
+
+function BillingSetupModal(props: { onClose: () => void }) {
+    const { isDark } = useContext(ThemeContext);
+    const [stripePromise, setStripePromise] = useState<Promise<Stripe | null> | undefined>();
+    const [stripeSetupIntentClientSecret, setStripeSetupIntentClientSecret] = useState<string | undefined>();
+
+    useEffect(() => {
+        const { server } = getGitpodService();
+        Promise.all([
+            server.getStripePublishableKey().then((v) => () => setStripePromise(loadStripe(v || ""))),
+            server.getStripeSetupIntentClientSecret().then((v) => () => setStripeSetupIntentClientSecret(v)),
+        ]).then((setters) => setters.forEach((s) => s()));
+    }, []);
+
+    const getStripeAppearance = (): Appearance => {
+        return {
+            theme: isDark ? "night" : "flat",
```

Comment:
**issue**: Although flat looks slightly better, should we default to the `stripe` theme so that UX looks similar with the rest of the product? 

| Variable Modal | Billing Modal |
|-|-|
| <img width="1440" alt="Screenshot 2022-06-07 at 5 39 49 PM (2)" src="https://user-images.githubusercontent.com/120486/172428363-473be223-7aee-4934-86d0-cee89113b6e4.png"> | <img width="1440" alt="Screenshot 2022-06-07 at 5 39 40 PM (2)" src="https://user-images.githubusercontent.com/120486/172428367-1fd6746a-085d-465f-8ff3-38e9bf23f0a3.png"> |